src/database/DataTypes/RequisitionItem.js

```diff
@@ -43,19 +44,21 @@ export class RequisitionItem extends Realm.Object {
     const { availableQuantity } = this.linkedTransactionItem;
     const { totalQuantity } = this.item;
 
-    return this.linkedTransactionItem ? availableQuantity : totalQuantity;
+    // TODO: this.linkedTransactionItem and this.item can both be falsey, unhandled possible return
+    if (availableQuantity) return availableQuantity;
+    return totalQuantity;
   }
 
   setSuppliedQuantity(database, newValue) {
     if (this.requisition.isFinalised || this.requisition.isRequest) {
       throw new Error('Cannot set supplied quantity for Finalised or Request Requisition');
     }
 
-    const transactionItem = this.linkedTransactionItem;
+    const { linkedTransactionItem: transactionItem } = this;
```

Comment:
As per our chart about destructuring this (I am not against it btw).

I can see you are reverting some of the destructuring of this, but the change here is unnecessary imo

Reply:
Reverted.